### PR TITLE
[TRUS-882] - Amend H1 header to include 'Lead'

### PR DESCRIPTION
**Pull request by jassalrichy**
[TRUS-882] - Amend H1 header to include 'Lead'

### CODE REVIEW

File: app/controllers/TrusteeAUKCitizenController.scala [hunk 1]
```diff
@@ -45,7 +46,7 @@ class TrusteeAUKCitizenController @Inject()(
                                              view: TrusteeAUKCitizenView
                                            )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
-  val form = formProvider()
+  var messagePrefix = ""
```
Comment (manishsw) [inline app/controllers/TrusteeAUKCitizenController.scala]: @jassalrichy  Remove if not required.

File: app/controllers/TrusteeIndividualOrBusinessController.scala [hunk 2]
```diff
@@ -47,7 +47,7 @@ class TrusteeIndividualOrBusinessController @Inject()(
   with I18nSupport
   with Enumerable.Implicits {
 
-  val form = formProvider()
+  var messagePrefix = ""
```
Comment (manishsw) [inline app/controllers/TrusteeIndividualOrBusinessController.scala]: same as above.

File: app/forms/TelephoneNumberFormProvider.scala [hunk 3]
```diff
@@ -23,12 +23,12 @@ import play.api.data.Form
...
-  def apply(): Form[String] =
+  def apply(messagePrefix: String): Form[String] =
     Form(
-      "value" -> text("telephoneNumber.error.required")
+      "value" -> text(s"$messagePrefix.error.required")
         .verifying(
           firstError(
-            isNotEmpty("value", "telephoneNumber.error.required"),
+            isNotEmpty("value", s"$messagePrefix.error.required"),
             regexp(Validation.telephoneRegex, "telephoneNumber.error.invalid.characters")
```
Comment (manishsw) [inline app/forms/TelephoneNumberFormProvider.scala]: @jassalrichy  is messagePrefix required here?

File: app/controllers/TrusteeIndividualOrBusinessController.scala (hunk 2 above)
Comment (jassalrichy) [reply app/controllers/TrusteeIndividualOrBusinessController.scala]: This is required

File: app/controllers/TrusteeAUKCitizenController.scala (hunk 1 above)
Comment (jassalrichy) [reply app/controllers/TrusteeAUKCitizenController.scala]: Removed as not required here

File: app/controllers/TelephoneNumberController.scala [hunk 4]
```diff
@@ -67,11 +70,26 @@ class TelephoneNumberController @Inject()(
       Ok(view(preparedForm, mode, index, trusteeName))
   }
 
+  private def getMessagePrefix(index: Int, request: DataRequest[AnyContent]) = {
+    val isLead = request.userAnswers.get(IsThisLeadTrusteePage(index)).get
```
Comment (adamconder) [inline app/controllers/TelephoneNumberController.scala]: @jassalrichy You're doing  a `.get` here on the Option without first checking it is defined.
Can we add a `requiredAnwer` to the actions and redirect if the answer does not exist?

File: app/controllers/TrusteeAUKCitizenController.scala [hunk 5]
```diff
@@ -84,4 +91,16 @@ class TrusteeAUKCitizenController @Inject()(
         }
       )
   }
+
+  private def getMessagePrefix(index: Int, request: DataRequest[AnyContent]) = {
+    val isLead = request.userAnswers.get(IsThisLeadTrusteePage(index)).get
```
Comment (adamconder) [inline app/controllers/TrusteeAUKCitizenController.scala]: @jassalrichy Same comment as above regarding the `.get`

File: app/controllers/TrusteesAnswerPageController.scala [hunk 6]
```diff
@@ -59,13 +59,18 @@ class TrusteesAnswerPageController @Inject()(
 
       val checkYourAnswersHelper = new CheckYourAnswersHelper(countryOptions)(request.userAnswers)
 
+      val isLead = request.userAnswers.get(IsThisLeadTrusteePage(index)).get
```
Comment (adamconder) [inline app/controllers/TrusteesAnswerPageController.scala]: Same with the `.get`

File: app/forms/TelephoneNumberFormProvider.scala (hunk 3 above)
Comment (adamconder) [reply app/forms/TelephoneNumberFormProvider.scala]: It depends if the error message is specific to the entity.

**Comment by adamconder** [comment]
@jassalrichy No further comments from me.

File: app/controllers/TelephoneNumberController.scala (hunk 4 above)
Comment (jassalrichy) [reply app/controllers/TelephoneNumberController.scala]: requiredAnswer check added

File: app/forms/TelephoneNumberFormProvider.scala (hunk 3 above)
Comment (jassalrichy) [reply app/forms/TelephoneNumberFormProvider.scala]: @manishsw required as error message is specific

File: app/controllers/TrusteeAUKCitizenController.scala (hunk 5 above)
Comment (jassalrichy) [reply app/controllers/TrusteeAUKCitizenController.scala]: requiredAnswer check added

File: app/controllers/TrusteesAnswerPageController.scala (hunk 6 above)
Comment (jassalrichy) [reply app/controllers/TrusteesAnswerPageController.scala]: requiredAnswer check added

**Pull request merged**